server/paymentProviders/transferwise/webhook.ts

```diff
@@ -117,42 +121,144 @@ export async function handleTransferStateChange(event: TransferStateChangeEvent)
   }
 }
 
+const handleTransferRefund = async (event: TransferRefundEvent): Promise<void> => {
+  const transferId = event.data.resource.id;
+  const refundWiseEventTimestamp = event.data.occurred_at;
+  const expense = await models.Expense.findOne({
+    where: {
+      status: [expenseStatus.PROCESSING, expenseStatus.PAID],
+      data: { transfer: { id: transferId } },
+    },
+    include: [
+      {
+        model: models.Collective,
+        as: 'collective',
+        include: [{ model: models.Collective, as: 'host', required: true }],
+        required: true,
+      },
+      { model: models.User, as: 'User' },
+      { model: models.Transaction },
+    ],
+  });
+
+  if (!expense) {
+    // This is probably some other transfer not executed through our platform.
+    logger.warn('Could not find related Expense, ignoring transferwise event.', event);
+    return;
+  } else if (expense.data.refundEventTimestamp === refundWiseEventTimestamp) {
+    logger.debug('Ignoring duplicate refund event.', event);
+    return;
+  }
+
+  const collective = expense.collective;
+  const host = collective.host;
+
+  const refundCurrency = event.data.resource.refund_currency;
+  if (refundCurrency !== expense.data.transfer.sourceCurrency) {
+    // This condition is guaranteed by Wise, but we should still check it
+    // Can we recover from this? How to infer the correct FX Rate so we know if this is a partial refund or not?
+    logger.warn('Refund currency does not match transfer source currency', event);
+    throw new Error('Refund currency does not match transfer source currency.');
+  }
+
+  const refundedAmount = event.data.resource.refund_amount;
+  const sourceAmount = expense.data.transfer.sourceValue;
+  const relatedTransferTransactions = expense.Transactions.filter(t => t.data?.transfer?.id === transferId);
+  const hasTransactions = relatedTransferTransactions.some(t => t.kind === TransactionKind.EXPENSE);
+
+  if (hasTransactions) {
+    assert.equal(refundCurrency, host.currency, 'Refund currency does not match host currency');
+    const creditTransaction = relatedTransferTransactions.find(
+      t => t.type === TransactionTypes.CREDIT && t.kind === TransactionKind.EXPENSE,
+    );
+    assert(creditTransaction, 'Could not find related CREDIT transaction');
+    const paymentProcessorFee = expense.data.paymentOption.fee.total;
+
+    if (refundedAmount === sourceAmount && expense.status === expenseStatus.PAID) {
+      logger.verbose('Wise: Paid Expense was fully refunded', event);
+      await createRefundTransaction(
+        creditTransaction,
+        paymentProcessorFee * 100,
+        pick(creditTransaction.data, ['transfer']),
+        expense.User,
+      );
+    } else if (refundedAmount < sourceAmount) {
+      logger.verbose('Wise: Paid Expense was partially refunded', event);
+      const difference = sourceAmount - refundedAmount;
+      const paymentProcessorFee = expense.data.paymentOption.fee.total;
```

Comment:
Can we end up with a difference greater than the payment processing fee?
Maybe when the `feesPayer = PAYEE`.